### PR TITLE
Hydroponics Cleanup

### PR DESCRIPTION
**Pull request by maybeserendipitous**
[CLEANLINESS][BUG]

## About the PR 

This cleans up some hydroponics related code, mainly the tray, seed manipulator, and plants in general.
It also includes some very minor bug fixes.

This also removes `harvest_tools`, `harvest_tool_message`, `harvest_tool_fail_message`, `required_reagents`, and related code.
In the interest of searchability, I will list the original excised fields and their associated comments here.

```
	var/list/harvest_tools // For plants that don't harvest normally and need some sort of special tool (mixed list of tool flags and item paths)
	var/harvest_tool_message // An output message for plants with unique harvest messages (string)
	var/harvest_tool_fail_message // A helpful output message to players when they attempt to harvest a plant by hand

	var/list/required_reagents // reagents required for the plant to grow - formated like: list(list(id="poo",amount=100),list(id="thing",amount=number))
```

Supersedes #9524 

## Why's this needed? 

Code cleanliness, possible performance improvements due to excising unneeded code.

### CODE REVIEW

File: code/modules/hydroponics/plants.dm [hunk 1]
```diff
@@ -35,11 +35,7 @@ ABSTRACT_TYPE(/datum/plant)
 	var/list/mutations = list() // what mutant variants does this plant have?
 	var/genome = 0 // Used for splicing - how "similar" the plants are = better odds of splice
 	var/stop_size_scaling // Stops the enlarging of sprites based on quality
-	var/list/harvest_tools // For plants that don't harvest normally and need some sort of special tool (mixed list of tool flags and item paths)
-	var/harvest_tool_message // An output message for plants with unique harvest messages (string)
-	var/harvest_tool_fail_message // A helpful output message to players when they attempt to harvest a plant by hand
 	var/no_extract // Stops the extraction of seeds in the PlantMaster
-	var/list/required_reagents // reagents required for the plant to grow - formated like: list(list(id="poo",amount=100),list(id="thing",amount=number))
```
Comment (ZeWaka) [inline code/modules/hydroponics/plants.dm:42]: what's with these removals? were they added elsewhere or something

Comment (maybeserendipitous) [reply code/modules/hydroponics/plants.dm:42]: Harvest tools etc. and required reagents were added two years ago but are never actually used by any plants (from what I've seen). I figured if it's been that long without even a single plant using them then we probably don't need them. If you want to keep them I don't mind undoing the removing commits.

Comment (ZeWaka) [reply code/modules/hydroponics/plants.dm:42]: hmm ok - can we make clear the removal in the PR description so people can find it easily to add it back if they want?

btw anything done by me in the 2 year ago initial commit is from the pre-open source code, and is not actually 2 years old

Comment (maybeserendipitous) [reply code/modules/hydroponics/plants.dm:42]: I listed the excised fields in the description, let me know if that works.

**Pull request merged**